src/StochasticStyles/spawning.jl

```diff
@@ -70,8 +70,8 @@ Perform diagonal step on a walker `add => val`. Optional argument `threshold` se
 projection threshold. If `eltype(w)` is an `Integer`, the `val` is rounded to the nearest
 integer stochastically.
 """
-@inline function diagonal_step!(w, op, add, val, threshold=0)
-    new_val = diagonal_element(op, add) * val
+@inline function diagonal_step!(w, column, add, val, threshold=0)
+    new_val = first(column).second*val
```

Comment:
I know this is the interface we talked about, but I'd actually prefer it if there was a `diagonal_element` function for doing this instead. Especially if we want to have operators that can't be iterated, defining `first` on them anyway is not the cleanest option.

Reply:
I agree. We should define 
```julia
diagonal_element(column) = first(column)[2]
```
and then use it whenever we need the diagonal element of an operator.